src/lib.rs

```diff
@@ -786,6 +792,58 @@ impl Account {
     pub fn id(&self) -> &str {
         &self.inner.id
     }
+
+    /// Account key rollover
+    pub async fn key_change(&self, server_url: &str) -> Result<AccountCredentials, Error> {
+        let new_key_url = match self.inner.client.directory.key_change.as_deref() {
+            Some(url) => url,
+            None => return Err("Account key rollover not supported by ACME CA".into()),
+        };
+
+        let (new_key, new_key_pkcs8) = Key::generate()?;
+
+        let jwk_old = Jwk::new(&self.inner.key.inner);
+
+        let payload_inner = NewKeyPayload {
+            account: self.inner.id.clone(),
+            old_key: jwk_old,
+        };
+
+        let mut inner_header = new_key.header(Some("nonce"), &new_key_url);
+        inner_header.nonce = None;
+
+        let inner_body = JoseJson::new(Some(&payload_inner), inner_header, &new_key)?;
+
+        let rsp = self
+            .inner
+            .post(Some(&inner_body), None, &new_key_url)
+            .await?;
+
+        let _ = Problem::from_response(rsp).await?;
+
+        let credentials = AccountCredentials {
+            id: self.inner.id.clone(),
+            key_pkcs8: new_key_pkcs8.as_ref().to_vec(),
+            directory: Some(server_url.to_owned()),
+            urls: None,
+        };
+
+        Ok(credentials)
+    }
+
+    /// Updates the account settings
+    pub async fn update_contact<'a>(&self, contact: &'a [&'a str]) -> Result<(), Error> {
```

Comment:
Let's rename this to `update_contacts()`, and rename the argument to `contacts`. Can you add some additional detail in a second paragraph in the docstring to clarify whether this extends or replaces the existing contacts?